rfc/system/3749-sane-overridable-defaults.md

```diff
@@ -0,0 +1,208 @@
+
+In FlyteAdmin config
+
+```
+flyteadmin:
+  overrides:
+    # Points to deprecated matchable overrides written by flytectl
+    sourceType: db 
+```
+OR 
+
+In FlyteAdmin config
+
+```
+flyteadmin:
+  overrrides:
+    sourceType: configmap
+    sourcePath: /etc/flyte/overrides/
+
+```
+
+And the corresponding ConfigMap which gets mounted as a volume mount in the FlyteAdmin deployment
+```
+kind: ConfigMap
+metadata:
+    name: defaults
+    namespace: flyte
+    apiVersion: v1
+    data:
+      defaults.yaml: |
+        # MyExampleWorkflow overrides
+        - project: flytesnacks
+          domain: development
+          workflow: MyExampleWorkflow
+          overrides:
+              task_resources:
+                  defaults:
+                      cpu: "1"
+                      memory: "500Mi"
+                  limits:
+                      cpu: "3"
+                      memory: "2Gi"
+                      gpu: "4"
+              ...         
+        # Flytesnacks/development level overrides            
+        - project: flytesnacks
+          domain: development
+          overrides:
+               workflow_execution_config:
+                    labels:
+                        my-key: "my-value"
+                    security_context:
+                        core_identity:
+                            iam_role: "my_iam_role"
+        ...                             
+
+```
+This can work the same way cluster resource templates are mounted as ConfigMaps in the FlyteAdmin deployment but are [fully customizable](https://github.com/flyteorg/flyte/blob/5ab8cb887a5a7051070cb93fca603ed1f22f2f74/charts/flyte-core/values.yaml#L752-L810).
+
+This does require restarting the FlyteAdmin deployment for invididual overrides changes but allows for declaratively defining the complete set of overrides in a reproducible manner. See more discusison on the trade-offs [here](https://github.com/flyteorg/flyte/discussions/3731#discussioncomment-6053743).
+
+#### FlyteAdmin changes
+
+##### Deprecated DB: Get Overrides
+Getting overrides at a super specific level, for example at the workflow level, should use the same fallback logic currently applied at execution time to determine all applicable overrides across all matchable resource types (task resources, workflow execution config, etc).
+
+That is, the workflow-level document should include all applicable workflow overrides, any project overrides and any domain overrides in one document with inherited overrides marked as such.
+
+##### Single ConfigMap
+When the FlyteAdmin application starts, it can load the overridable defaults configmap and create an in-memory the mapping of per-project, per-domain and per-workflow overrides.
```

Comment:
```suggestion
When the FlyteAdmin application starts, it can load the overridable defaults configmap and create an in-memory mapping of per-project, per-domain and per-workflow overrides.
```

Reply:
done, thanks